tests/cupy_tests/lib_tests/test_polynomial.py

```diff
@@ -238,3 +259,39 @@ def test_poly1d_ne1(self, xp, dtype):
     def test_poly1d_ne2(self, xp, dtype):
         a, b = self.make_poly1d2(xp, dtype)
         return a != b
+
+
+@testing.gpu
+class TestPoly1dDeviceSynchronization(unittest.TestCase):
```

Comment:
How about adding `cupyx.allow_synchronize(False)` to `TestPoly1dInit` instead of this class?